docs/developers/testing.rst

```diff
@@ -0,0 +1,110 @@
+==================
+Run MicroOVN tests
+==================
+
+MicroOVN has two types of tests, linter checks and functional tests and this
+page will show how to run them.
+
+Linter checks
+-------------
+
+Linting is currently applied only to the tests themselves, not the main
+codebase. The prerequisites for running linter are:
+
+* ``make``
+* ``shellcheck``
+
+You can install them with:
+
+.. code-block:: none
+
+   sudo apt install make shellcheck
+
+To perform linting, go into the repository's root directory and run:
+
+.. code-block:: none
+
+   make check-lint
+
+Functional tests
+----------------
+
+These tests build the MicroOVN snap and use it to deploy the ``OVN`` cluster
```

Comment:
i'm thrown by the "build" part here. the other how-to showed how to build. it seems that for testing, a snap is built and auto-deployed in containers but it feels off since:

a) we seem to be redoing stuff that may already be done
b) it's not clear from what source branch MicroOVN is being built from (and being tested)